src/Native/ObjWriter/objwriter.cpp

```diff
@@ -333,15 +341,28 @@ void ObjectWriter::EmitSymbolDef(const char *SymbolName, bool global) {
     Streamer->EmitSymbolAttribute(Sym, MCSA_Local);
   }
 
+  Triple TheTriple = TMachine->getTargetTriple();
+
   // A Thumb2 function symbol should be marked with an appropriate ELF
   // attribute to make later computation of a relocation address value correct
-  if (GetTriple().getArch() == Triple::thumb &&
-      GetTriple().getObjectFormat() == Triple::ELF &&
+
+  if (TheTriple.getObjectFormat() == Triple::ELF &&
       Streamer->getCurrentSectionOnly()->getKind().isText()) {
-    Streamer->EmitSymbolAttribute(Sym, MCSA_ELF_TypeFunction);
+      switch (TheTriple.getArch()) {
+        case Triple::thumb:
+        case Triple::aarch64:
+          Streamer->EmitSymbolAttribute(Sym, MCSA_ELF_TypeFunction);
+          break;
+
+        default:
+          break;
+    }
   }
 
-  Streamer->EmitLabel(Sym);
+  if (Sym->isUndefined())
```

Comment:
Why was this needed? Is the compiler trying to define the same symbol twice somewhere? (We should have asserts on the managed side that this doesn't happen.)

Reply:
It isn't. It's a left over. I will remove it